packages/server/src/restaurants/profile/domain/UpdatePartnerProfileHandler.ts

```diff
@@ -20,19 +21,27 @@ class UpdatePartnerProfileHandler implements Handler<UpdatePartnerProfileRequest
   constructor(@InjectModel(Restaurant.name) private restaurantModel: Model<RestaurantDocument>) {}
 
   async exec(req: UpdatePartnerProfileRequest): Promise<null | undefined> {
-    const result = await this.restaurantModel.findOneAndUpdate(
-      { _id: req.id },
-      {
-        name: req.name,
-        description: req.description,
-        cuisineType: req.cuisineType,
-        tags: req.tags,
-        bankAccountNumber: req.bankAccountNumber,
-        phoneNumber: req.phoneNumber,
-      },
-    );
+    const filter = { _id: req.id };
+    const update = {
+      name: req.name ? req.name : null,
+      description: req.description ? req.description : null,
+      cuisineType: req.cuisineType ? req.cuisineType : null,
+      tags: req.tags ? req.tags : null,
+      bankAccountNumber: req.bankAccountNumber ? req.bankAccountNumber : null,
+      phoneNumber: req.phoneNumber ? req.phoneNumber : null,
+    };
```

Comment:
```suggestion
    const update = {
      name: req.name || null,
      description: req.description || null,
      cuisineType: req.cuisineType || null,
      tags: req.tags || null,
      bankAccountNumber: req.bankAccountNumber || null,
      phoneNumber: req.phoneNumber || null,
    };
```

Chyba że sprawdzasz tylko czy jest undefined/null a nie ogólnie falsy, to `??` będzie lepsze od `||`.